src/AutoRest.CSharp/Common/Output/Expressions/ExtensibleSnippets.ArgumentSnippets.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.CSharp.Common.Output.Expressions.Statements;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Output.Models.Shared;
+using static AutoRest.CSharp.Common.Output.Models.Snippets;
+
+namespace AutoRest.CSharp.Common.Output.Expressions
+{
+    internal abstract partial class ExtensibleSnippets
+    {
+        internal abstract class ArgumentSnippets
```

Comment:
Why is this an inner class instead of its own file?

Reply:
This does not have particular reason, just a way Alexander chose to have them well organized.
It just works the same if we choose to have those individual types and put them inside ExtensibleSnippets namespace.
No one should use the class name directly - the expression `Snippets.Extensible.Argument` returns you an singleton of this class.